### ISSUE TITLE
Error uploading a XLSX file

### ISSUE THREAD

**Issue by uliss3s** [ckan/ckan-docker#62]
Just downloaded the repo, changed the env variable CKAN_SITE_URL, and run "docker compose up -d"

Trying to upload a XLSX file results in the error bellow.
The resource datastore status stays on "Pending" forever.
```
datapusher  | 2023-06-19 13:46:43,073 ERROR Job "push_to_datastore (trigger: date[2023-06-19 13:46:43 UTC], next run at: 2023-06-19 13:46:43 UTC)" raised an exception
datapusher  | Traceback (most recent call last):
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/excel.py", line 43, in get_workbook
datapusher  |     return xlrd.open_workbook(
datapusher  |   File "/usr/lib/python3.9/site-packages/xlrd/__init__.py", line 170, in open_workbook
datapusher  |     raise XLRDError(FILE_FORMAT_DESCRIPTIONS[file_format]+'; not supported')
datapusher  | xlrd.biffh.XLRDError: Excel xlsx file; not supported
datapusher  | 
datapusher  | During handling of the above exception, another exception occurred:
datapusher  | 
datapusher  | Traceback (most recent call last):
datapusher  |   File "/usr/lib/python3.9/site-packages/datapusher-0.0.20-py3.9.egg/datapusher/jobs.py", line 433, in push_to_datastore
datapusher  |     table_set = messytables.any_tableset(tmp, mimetype=ct, extension=ct)
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/any.py", line 137, in any_tableset
datapusher  |     return parsers[attempt](fileobj, **kw)
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/excel.py", line 78, in __init__
datapusher  |     self.workbook = get_workbook()
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/excel.py", line 53, in get_workbook
datapusher  |     raise ReadError("Can't read Excel file: %r" % value).with_traceback(traceback)
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/excel.py", line 43, in get_workbook
datapusher  |     return xlrd.open_workbook(
datapusher  |   File "/usr/lib/python3.9/site-packages/xlrd/__init__.py", line 170, in open_workbook
datapusher  |     raise XLRDError(FILE_FORMAT_DESCRIPTIONS[file_format]+'; not supported')
datapusher  | messytables.error.ReadError: Can't read Excel file: XLRDError('Excel xlsx file; not supported')
datapusher  | 
datapusher  | During handling of the above exception, another exception occurred:
datapusher  | 
datapusher  | Traceback (most recent call last):
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/excel.py", line 43, in get_workbook
datapusher  |     return xlrd.open_workbook(
datapusher  |   File "/usr/lib/python3.9/site-packages/xlrd/__init__.py", line 170, in open_workbook
datapusher  |     raise XLRDError(FILE_FORMAT_DESCRIPTIONS[file_format]+'; not supported')
datapusher  | xlrd.biffh.XLRDError: Excel xlsx file; not supported
datapusher  | 
datapusher  | During handling of the above exception, another exception occurred:
datapusher  | 
datapusher  | Traceback (most recent call last):
datapusher  |   File "/usr/lib/python3.9/site-packages/datapusher-0.0.20-py3.9.egg/datapusher/jobs.py", line 439, in push_to_datastore
datapusher  |     table_set = messytables.any_tableset(tmp, mimetype=format, extension=format)
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/any.py", line 146, in any_tableset
datapusher  |     return parsers[attempt](fileobj, **kw)
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/excel.py", line 78, in __init__
datapusher  |     self.workbook = get_workbook()
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/excel.py", line 53, in get_workbook
datapusher  |     raise ReadError("Can't read Excel file: %r" % value).with_traceback(traceback)
datapusher  |   File "/usr/lib/python3.9/site-packages/messytables/excel.py", line 43, in get_workbook
datapusher  |     return xlrd.open_workbook(
datapusher  |   File "/usr/lib/python3.9/site-packages/xlrd/__init__.py", line 170, in open_workbook
datapusher  |     raise XLRDError(FILE_FORMAT_DESCRIPTIONS[file_format]+'; not supported')
datapusher  | messytables.error.ReadError: Can't read Excel file: XLRDError('Excel xlsx file; not supported')
datapusher  | 
datapusher  | During handling of the above exception, another exception occurred:
datapusher  | 
datapusher  | Traceback (most recent call last):
datapusher  |   File "/usr/lib/python3.9/site-packages/apscheduler/executors/base.py", line 125, in run_job
datapusher  |     retval = job.func(*job.args, **job.kwargs)
datapusher  |   File "/usr/lib/python3.9/site-packages/datapusher-0.0.20-py3.9.egg/datapusher/jobs.py", line 441, in push_to_datastore
datapusher  |     raise util.JobError(e)
datapusher  | ckanserviceprovider.util.JobError: <unprintable JobError object>
datapusher  | 2023-06-19 13:46:43,077 ERROR Error notifying listener
datapusher  | Traceback (most recent call last):
datapusher  |   File "/usr/lib/python3.9/site-packages/apscheduler/schedulers/base.py", line 835, in _dispatch_event
datapusher  |     cb(event)
datapusher  |   File "/usr/lib/python3.9/site-packages/ckanserviceprovider/web.py", line 200, in job_listener
datapusher  |     db.mark_job_as_errored(job_id, error_object)
datapusher  |   File "/usr/lib/python3.9/site-packages/ckanserviceprovider/db.py", line 419, in mark_job_as_errored
datapusher  |     _update_job(job_id, update_dict)
datapusher  |   File "/usr/lib/python3.9/site-packages/ckanserviceprovider/db.py", line 354, in _update_job
datapusher  |     job_dict["error"] = _validate_error(job_dict["error"])
datapusher  |   File "/usr/lib/python3.9/site-packages/ckanserviceprovider/db.py", line 329, in _validate_error
datapusher  |     raise InvalidErrorObjectError("error['message'] must be a string")
datapusher  | ckanserviceprovider.db.InvalidErrorObjectError: error['message'] must be a string
```

**Comment by kowh-ai** [comment]
Yes. Take a look at [this datapusher issue](https://github.com/ckan/datapusher/issues/232)

**Comment by uliss3s** [comment]
It's a datapusher issue

**Comment by tazchalmers** [comment]
Hi.
I found this closed post as I have hit the same issue with the same error when uploading an xlsx file in CKAN v2.10.4. However, the datapusher documentation https://docs.ckan.org/en/2.10/maintaining/configuration.html#ckan-datapusher-formats states that xlsx is supported:

`ckan.datapusher.formats
Default value: csv xls xlsx tsv application/csv application/vnd.ms-excel application/vnd.openxmlformats-officedocument.spreadsheetml.sheet ods application/vnd.oasis.opendocument.spreadsheet`

Should this be revised?

I found reference to datapusher-plus as a possible alternative that does support xlsx. Is there a plan to package this in the next stable release of the dockerised version?
